pkg/pipeline/BuildPipelineConfigService.go

```diff
@@ -2007,8 +2011,8 @@ func (impl *CiPipelineConfigServiceImpl) DeleteCiPipeline(request *bean.CiPatchR
 	ciPipelineId := request.CiPipeline.Id
 
 	// check for linked ci's before deleting the ci pipeline
-	count, err := impl.ciPipelineRepository.FindByLinkedCiCount(ciPipelineId)
-	if err != nil && err != pg.ErrNoRows {
+	count, err := impl.ciPipelineConfigReadService.GetChildrenCiCount(ciPipelineId)
+	if err != nil {
```

Comment:
ensure errNoRows case is handled

Reply:
handled inside `GetChildrenCiCount` func